lib/index.js

```diff
@@ -174,9 +170,7 @@ const fetch = (url, opts) => {
           }
 
           // Update host due to redirection
-          // disabling no-deprecated-api pending refactor to URL constructor
-          // eslint-disable-next-line node/no-deprecated-api
-          request.headers.set('host', Url.parse(locationURL).host)
+          request.headers.set('host', (new URL(locationURL)).host)
```

Comment:
on line 128 we're allowing `locationURL` to be `null` sometimes, which would make this explode.